### ISSUE TITLE
`ddev phpcs` reports failures unrelated to module code

### ISSUE THREAD

**Issue by jameswilson** [ddev/ddev-drupal-contrib#52]
## Problem statement

`ddev phpcs` appears to be scanning the .ddev folder, and finds issues with .ddev/xhprof/xhprof_prepend.php.  

```

FILE: /var/www/html/web/modules/custom/svg/.ddev/xhprof/xhprof_prepend.php
-------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AND 2 WARNINGS AFFECTING 3 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------------------
 22 | WARNING | Line exceeds 80 characters; contains 85 characters (Drupal.Files.LineLength.TooLong)
 41 | WARNING | Line exceeds 80 characters; contains 92 characters (Drupal.Files.LineLength.TooLong)
 48 | ERROR   | Doc comment short description must end with a full stop (Drupal.Commenting.DocComment.ShortFullStop)
 48 | ERROR   | Doc comment short description must be on a single line, further text should be a separate paragraph (Drupal.Commenting.DocComment.ShortSingleLine)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 462ms; Memory: 6MB


PHP CODE SNIFFER REPORT SUMMARY
-----------------------------------------------------------------------------------------
FILE                                                                     ERRORS  WARNINGS
-----------------------------------------------------------------------------------------
/var/www/html/web/modules/custom/svg/.ddev/xhprof/xhprof_prepend.php     2       2
-----------------------------------------------------------------------------------------
A TOTAL OF 2 ERRORS AND 2 WARNINGS WERE FOUND IN 1 FILE
-----------------------------------------------------------------------------------------


PHP CODE SNIFFER VIOLATION SOURCE SUMMARY
----------------------------------------------------------------------
SOURCE                                                           COUNT
----------------------------------------------------------------------
Drupal.Files.LineLength.TooLong                                  2
Drupal.Commenting.DocComment.ShortFullStop                       1
Drupal.Commenting.DocComment.ShortSingleLine                     1
----------------------------------------------------------------------
A TOTAL OF 4 SNIFF VIOLATIONS WERE FOUND IN 3 SOURCES
----------------------------------------------------------------------
```

## Proposed resolution

I suppose technically we could open an upstream issue to fix this in Ddev's repo, but the file appears to be written with non-Drupal PHP standards, so it might not be accepted and `ddev phpcbf` would then continue to incorrectly try to apply Drupal formatting to the file anyway.

Should we instead open an issue to exclude `.ddev/xhprof` folder in 

https://git.drupalcode.org/project/gitlab_templates/-/blob/main/assets/phpcs.xml.dist 

Open to other approaches

**Comment by rfay** [comment]
Related:
* https://github.com/ddev/ddev/issues/6173

IMO it would be fine to exclude the .ddev directory, and also a PR to make that compliant would be fine.

As was done the other day on 
* https://github.com/ddev/ddev/pull/6217

**Comment by weitzman** [comment]
Fixed earlier today in #51

**Comment by joelpittet** [comment]
I ran into this too, but on GitlabCI, this was my workaround:
https://git.drupalcode.org/project/calendar/-/merge_requests/54/diffs?commit_id=92f75a4dd60c3d19e0b21a503fd511e15eacc08e

```
  <!-- Exclude the .ddev directory -->
  <exclude-pattern>**/.ddev/*</exclude-pattern>
```

**Comment by jameswilson** [comment]
@fjgarlin What do you think about adding the following line to the [GitLab phpcs.xml.dist template](https://git.drupalcode.org/project/gitlab_templates/-/raw/default-ref/assets/phpcs.xml.dist) to ignore false positives in the .ddev directory?

```xml
  <!-- Exclude the .ddev directory -->
  <exclude-pattern>**/.ddev/*</exclude-pattern>
```

This should apply to both GitLab CI and local `ddev phpcs`, since they use the same config file.

**Comment by fjgarlin** [comment]
Would this still be needed? I think the fix makes more sense here (done in #51) given that it is a side effect of using this tool for testing.

Having said that, I know that this plugin is actually the recommended option to test locally, so I'd be ok if an issue is open against the `gitlab_templates` project to perhaps add the option to remove any `.ddev/*` files when testing the project in GitLab CI, rather than fixing nuances job by job.
